src/components/Hoverable/index.js

```diff
@@ -1,147 +1,93 @@
 import _ from 'underscore';
-import React, {Component} from 'react';
+import React, {useEffect, useCallback, useState, useRef, useMemo, useImperativeHandle} from 'react';
 import {propTypes, defaultProps} from './hoverablePropTypes';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 
+function mapChildren(children, callbackParam) {
+    if (_.isArray(children) && children.length === 1) return children[0];
+
+    if (_.isFunction(children)) return children(callbackParam);
+
+    return children;
+}
+
 /**
  * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
  * because nesting Pressables causes issues where the hovered state of the child cannot be easily propagated to the
  * parent. https://github.com/necolas/react-native-web/issues/1875
  */
-class Hoverable extends Component {
-    constructor(props) {
-        super(props);
-
-        this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
-        this.checkHover = this.checkHover.bind(this);
-
-        this.state = {
-            isHovered: false,
-        };
-
-        this.wrapperView = null;
-    }
-
-    componentDidMount() {
-        document.addEventListener('visibilitychange', this.handleVisibilityChange);
-        document.addEventListener('mouseover', this.checkHover);
```

Comment:
@kacper-mikolajczak It seems this listener has been not migrated, `checkHover` is not needed anymore?

Reply:
Great catch, thanks! Done ✅